src/schema/users.ts

```diff
@@ -567,6 +648,44 @@ export const resolvers: IResolvers<any, Context> = {
         )}.png?r=${randomStr}`,
       };
     },
+    updateUserProfile: async (
+      _,
+      { data }: GQLUserParameters,
+      ctx,
+    ): Promise<GQLUser> => {
+      const repo = ctx.con.getRepository(User);
+      const user = await repo.findOne({ id: ctx.userId });
+
+      if (!user) {
+        throw new AuthenticationError(
+          JSON.stringify({ auth: 'User not found' }),
+        );
+      }
+
+      if (data.username && user.username != data.username) {
+        const takenUsername = await repo.findOne({ username: data.username });
+        if (takenUsername) {
+          throw new ValidationError(
+            JSON.stringify({ username: 'Username is already taken' }),
+          );
+        }
+      }
+
+      const links = ['twitter', 'github', 'hashnode', 'portfolio'];
```

Comment:
I think that twitter and github are not stored as links.
It's better to copy the validation logic from the gateway

Reply:
Ah, thanks for the tip. Let me check that one.

Reply:
Ah, thanks for the tip. Let me check that one. Though we use a library for validating the body, should we install it here as well?

Reply:
No but we can use the same logic I guess